### PR TITLE
Add predicate to removeAll

### PR DESCRIPTION
**Pull request by mackoj**
Hi @kean,

I have an issue I need to clean the log from private information before sharing it. I don't see a way to do this so I did this PR to try to implement it or at least provide an idea of how to do it.

I want to be able to remove all LoggerMessageEntity that contain certain pieces of information.

### CODE REVIEW

**Comment by mackoj** [comment]
I don't really know how to handle the **blob** ? Should I delete them ? or not ?

**Comment by mackoj** [comment]
At this current stage this patch block our app please to not merge, I still tinkering with it.

File: Sources/PulseCore/LoggerStore.swift [hunk 1]
```diff
@@ -541,14 +541,16 @@ extension LoggerStore {
...
 
-    private func _removeAll() {
+    private func _removeAll(_ predicate: NSPredicate? = nil) {
         switch document {
         case .directory(let blobs):
-            try? deleteMessages(fetchRequest: LoggerMessageEntity.fetchRequest())
+            let fetch = LoggerMessageEntity.fetchRequest()
+            if let predicate = predicate { fetch.predicate = predicate }
+            try? deleteMessages(fetchRequest: fetch)
             blobs.removeAll()
```
Comment (kean) [inline Sources/PulseCore/LoggerStore.swift:554]: This line removes all network responses.

Comment (mackoj) [reply Sources/PulseCore/LoggerStore.swift:554]: Ok how can I remove all non network response that match a predicate ?

Comment (mackoj) [reply Sources/PulseCore/LoggerStore.swift:554]: I want to filter normal log information not the network logs.

> If you want to filter out private information before sharing

Exactly

**Pull request closed without merging**